server/app.js

```diff
@@ -194,7 +194,9 @@ app.get('/download/:id', asyncHandler(async (req, res)=>{
 app.get('/user/:username', async (req, res, next)=>{
 	const ownAccount = req.account && (req.account.username == req.params.username);
 
-	let brews = await HomebrewModel.getByUser(req.params.username, ownAccount)
+	const fields = 'title pageCount description authors views shareId editId createdAt updatedAt lastViewed';

```

Comment:
Let's use the "array" syntax for this, to make it easier to add/remove values to `fields`  in the future.

See: https://mongoosejs.com/docs/api.html#query_Query-select

```
// include a and b, exclude other fields
query.select('a b');
// Equivalent syntaxes:
query.select(['a', 'b']);
```

Reply:
Otherwise I think this all looks good. 👍

Reply:
It's easy enough as a text string to just have it be `null` when not used, and then everything works as it has previously - I'll need to take a deeper dive into the `select` function to switch to that method.
Another option would be to use `fields = ['title','description'].join(' ');` to continue using the same parameter & function as it currently does.

Reply:
It should work just the same according to the Mongoose docs. Null should also work. No need to change anything else.

Reply:
Looks like the Mongoose select function can use a text string or list without any difficulty, so it made switching to the select function much easier. The latest commit should be using the new select function and a list of fields to return.

Reply:
Oh, LOL. I see your confusion. The code I linked to wasn't so you would use the Select functions I was just showing you the part of the docs that says you can use an array instead of a string. I apologize.

Can you please return it back to how you had it (not using Select), but just use an array?

Reply:
I've reverted to the find parameter method rather than the select function in the latest commit - to my mind, including it in the original function call *should* be more efficient than generating a full set of results and filtering it in a subsequent function.
I've no indication that this is actually the case, but all other things being equivalent, that's my preference.